### PR TITLE
APIGW NG add AWS_PROXY integration

### PR DESCRIPTION
**Pull request by bentsku**
## Motivation
This PR implements the `AWS_PROXY` API Gateway integration. 
https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html

## Changes
- populate the `identity` field of the context with the default values for a method with no authorizer
- port a few utils needed for the lambda integration
- implement the `AWS_PROXY` integration, adding the utils to create the input payload and parsing the response from lambda and passing it down the chain

## Testing
All the tests related to `AWS_PROXY` in `tests.aws.services.apigateway.test_apigateway_lambda` are passing when forcing the provider, with way less snapshot ignore than before 🥳 
(With `test_lambda_aws_proxy_integration_non_post_method` depending #11183 to be merged due to the gateway response default value issue).

## TODO

What's left to do:

- [ ] write maybe a bit of unit test
- [ ] manage all exceptions, do validation in console too
- [ ] ?

### CODE REVIEW

File: localstack-core/localstack/services/apigateway/next_gen/execute_api/helpers.py [hunk 1]
```diff
@@ -53,3 +55,36 @@ def render_uri_with_path_parameters(uri: str, path_parameters: dict[str, str]) -
...
+    )
+
+
+def get_lambda_function_arn_from_invocation_uri(uri: str) -> str:
+    """
+    "arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:123456789012:function:SimpleLambda4ProxyResource/invocations",
+    :param uri: the integration URI value for a lambda function
+    :return: the lambda function ARN
+    """
+    return uri.split(":lambda:path")[1].split("functions/")[1].removesuffix("/invocations")
```
Comment (cloutierMat) [inline localstack-core/localstack/services/apigateway/next_gen/execute_api/helpers.py:78]: It seems like you are doing an extra split? Wouldn't splitting once at `"functions/"` be enough?

File: localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/aws.py [hunk 2]
```diff
@@ -37,3 +87,173 @@ class RestApiAwsProxyIntegration(RestApiIntegration):
...
+            Payload=event,
+            InvocationType="Event" if asynchronous else "RequestResponse",
+        )
+        if payload := inv_result.get("Payload"):
+            return payload.read()
+        return b""
+
+    def parse_lambda_response(self, payload: bytes) -> LambdaProxyResponse:
+        try:
+            lambda_response = json.loads(payload)
```
Comment (cloutierMat) [inline localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/aws.py:175]: Is there still a weird edge case where a lambda returning `"[]"` would lead to a `TypeError` down the line?

File: tests/aws/services/apigateway/test_apigateway_lambda.py [hunk 3]
```diff
@@ -109,7 +89,15 @@ def test_lambda_aws_proxy_integration(
...
-    snapshot.add_transformer(snapshot.transform.key_value("deploymentId"))
+    # TODO: update global transformers, but we will need to regenerate all snapshots at once
+    snapshot.add_transformers_list(
+        [
+            snapshot.transform.key_value("deploymentId"),
+            snapshot.transform.jsonpath("$..headers.Host", value_replacement="host"),
+            snapshot.transform.jsonpath("$..multiValueHeaders.Host[0]", value_replacement="host"),
+        ],
+        priority=-1,
+    )
```
Comment (cloutierMat) [inline tests/aws/services/apigateway/test_apigateway_lambda.py:100]: That already cleaned it up a lot!
:rocket:

File: tests/aws/services/apigateway/test_apigateway_lambda.py [hunk 4]
```diff
@@ -300,8 +288,15 @@ def test_lambda_aws_proxy_integration_non_post_method(
...
     )
-    aws_client.apigateway.create_deployment(restApiId=api_id, stageName=stage_name)
 
+    # TODO: comment
+    aws_client.apigateway.put_gateway_response(
+        restApiId=api_id,
+        responseType="INTEGRATION_FAILURE",
+        responseParameters={},
+    )
+
```
Comment (cloutierMat) [inline tests/aws/services/apigateway/test_apigateway_lambda.py:298]: That seems like a left over from our session? Do you want to test some gateway responses here?

**Review by cloutierMat — COMMENTED** [review]
:rocket:  Super clean work and a nice set of utils we should be able to reuse!

Appart from a couple little nitpick there is the weird issue where the response could be an array, which would decode to json but fail down the line. It would probably be good to understand AWS behaviour here. I would assume It should raise the `InternalServerError` but it is hard to predict sometimes :rofl:

**Review by cloutierMat — APPROVED** [review]
Wow, it actually behave as expected when passing a list? :rofl: 
Great work :rocket:
